### DISCUSSION TITLE
Summay of Browser Redering Process

### DISCUSSION THREAD

**Discussion by honaseung** [honaseung/react-deep-dive#15] [Side dish]
I'm not gonna tell you whole process of How 'Browser Redering Process' works. I will tell you summarized process (referenced from book). So If you want detail, Visit [here](https://developer.mozilla.org/en-US/docs/Web/Performance/How_browsers_work).

### So are you ready to go? Here are the things
 1. Download `HTML`.
 2. Parsing `HTML` contents and making tree (DOM).
 3. Download `CSS`
 4. Parsing `CSS` contents and making tree (CSSOM).
 5. Analyzing `DOM` except `display:none` contents.
 6. With Result of step 5, Search `CSSOM` and apply it for 2 steps.
   6-1. Calutaing position.
   6-2. Painting colors.
![browser rendering process](https://github.com/honaseung/react-deep-dive/assets/53066347/078f85f3-28df-4199-8804-71c9f8fde89e)

**Comment by chaminju927** [comment]
Then, that picture would be include only repainting process, without layout/reflowing process!
